Reshaping Contributing docs (#67785)
WordPress/gutenberg: storybook/stories/contributing/contributing-documentation.txt

import { Meta } from '@storybook/addon-docs';

<Meta title="Contributing/Documentation">

This website is composed of articles with guidance on how we create products, which we call documentation.

Our components and patterns need detailed documentation for usage, style, code, and accessibility — that way everyone can easily use them to do great work.

We’d love for you to help us improve our documentation. It’s easy to get started.

*Is this your first contribution? You’ll first need to follow <Link story="Contributing/Get Ready to Contribute">a couple of steps to get started</Link>.*

### **Small copy changes**

> If you see a spelling, punctuation, or grammatical error anywhere in the documentation, you can help fix it. 

Reviewer: These trivial edits definitely don't need an issue opened beforehand — it can go direct to a PR.

Author: Opening an issue is easier than creating a PR for people who aren't comfortable with Git — especially in the current workflow that requires forking and cloning the entire repo. I think having this option allows more people to contribute, but I'll rewrite everything to make it clear that if you're comfortable creating a PR, that you can just do that :)

Ideally there'd be an "edit page" button that allows someone to directly create a PR with any copy edits, but I don't think that's possible with the current setup.

Reviewer: It should be pretty easy to make copy edits without forking/cloning locally. Edits can go straight to PR in the browser (for example using [this link](https://github.com/WordPress/gutenberg/edit/trunk/storybook/stories/tokens/color.mdx)).

> If you see a spelling, punctuation, or grammatical error anywhere in our documentation, you can help us fix it. 

Reviewer 2: ```suggestion
If you see a spelling, punctuation, or grammatical error anywhere in the documentation, you can help fix it. 
```

> By creating an issue on GitHub, you can easily suggest one or multiple small copy changes, which will then be approved and added by the community. In the near future, we’ll also have an Edit button on every page to instantly suggest changes to the page you’re on. We’ll update this page when this feature is live.

Reviewer 2: ```suggestion
You can suggest one or multiple small copy changes in a GitHub issue, which will then be approved and added by the community.
```

Reviewer 2: I'd stick to what is true at the moment, in case we end up not being able to include this feature.

[Click here to open a new issue](https://github.com/WordPress/gutenberg/issues/new?labels=Design+System&body=Then%2C+write+a+detailed+description%3A%0D%0A%0D%0A-+Explain+your+rationale+for+adding+this+component%0D%0A-+Detail+the+intended+behavior%0D%0A-+Clarify+if+it%E2%80%99s+a+new+component+of+a+variation+of+an+existing+one%0D%0A-+Include+a+basic+mockup+%28optional%29%0D%0A-+Include+inspiration+from+other+products+%28optional%29). Then, write a detailed description that explains the change(s) you’re making to the writing.

After opening your issue, reviewers will start a discussion in the comments to see if it's an appropriate change. When it’s approved, the community will take care of making the change — unless you want to create a PR yourself.

> ### **Adding new pages**

Reviewer: I'm kind of doubting that a new contributor would be able to (or even want to) author a page from scratch on a foundational aspect of the design system, which requires a deep understanding of the whole.

I wonder if we should focus on improving flows for the more likely entry points for new contributors, which will be to edit existing pages. For example I could add some kind of footer on the doc pages, that links out to a GitHub view to edit the doc.

If you’re adding a new component or pattern to the design system, or are adding other kinds of information, you’ll need to create a new page.

Creating a new page takes more work and will need feedback from the community. That’s why you’ll need to open an issue on GitHub to track the progress of your new page.

> #### **Step 1: Open a GitHub issue**

Reviewer 3: I was surprised to see this document going into such depth. Before going into writing those up in such detail, I'd suggest: 

* Referring to any pre-existing documentation in https://github.com/WordPress/gutenberg/tree/trunk/docs (in this case, https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors#repository-management and https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/repository-management.md#issues are good examples)
* Linking to the respective documentation from here, keeping this mostly an index with context and links
* Contributing to the existing `/docs` to improve them if a detail is missing
* Creating new `/docs` if an aspect of contributing is not documented at all

After all, if a document tries to address all questions and solve all problems, it usually addresses none. And there's no need to rediscover the wheel; we are standing on the shoulders of giants when reusing the existing docs that are the product of years of iteration from many contributors.

> If you’ve decided you want to create a new page, you’ll need to open a GitHub issue on Project Gutenberg. This is where we’ll be able to discuss the proposed changes and track progress.

Reviewer 2: ```suggestion
If you want to create a new page, you’ll need to open a GitHub issue on Project Gutenberg. This is where the discussion of the proposed changes and progress tracking will happen.
```

It’s encouraged to open an issue even if you can’t complete every part yourself, as someone in the community might be able to pick up where you left off.

[Click here to open a new issue](https://github.com/WordPress/gutenberg/issues/new?labels=Design+System&body=Then%2C+write+a+detailed+description%3A%0D%0A%0D%0A-+Explain+your+rationale+for+adding+this+component%0D%0A-+Detail+the+intended+behavior%0D%0A-+Clarify+if+it%E2%80%99s+a+new+component+of+a+variation+of+an+existing+one%0D%0A-+Include+a+basic+mockup+%28optional%29%0D%0A-+Include+inspiration+from+other+products+%28optional%29).

Then, write a detailed description:

* Explain your rationale for adding this page  
* Describe why this content hasn’t been covered yet in the design system

After opening your issue, reviewers will start a discussion in the comments to see if it's an appropriate addition, or if there's overlap with an existing page. They will also discuss what would be the best place for your place to live in the design system.

#### **Step 2: Write, share, refine**

> When your idea has been approved by the community, it’s time to start writing. We recommend doing this in [Google Docs](http://docs.google.com), as it’s easy to leave suggestions or comments on your writing.

Reviewer 2: ```suggestion
Once the community approves your idea, it’s time to start writing. We recommend doing this in [Google Docs](http://docs.google.com), as it’s easy to leave suggestions or comments on your writing.
```

> When your idea has been approved by the community, it’s time to start writing. We recommend doing this in [Google Docs](http://docs.google.com), as it’s easy to leave suggestions or comments on your writing.

Reviewer 2: > We recommend doing this in [Google Docs](http://docs.google.com), as it’s easy to leave suggestions or comments on your writing.

I would remove this from these docs and save it for internal documentation. Each contributor has a different way of working and not everyone has access to all GSuites

Make sure to [use Markdown in your writing](https://www.markdownguide.org/cheat-sheet/) so it can easily be moved to the documentation website when it’s done. To enable Markdown in Google Docs, click Tools \> Preferences \> Enable Markdown.

> It’s important that your writing aligns with the [Make WordPress Style Guide](https://make.wordpress.org/docs/style-guide/). You’ll find guidance on lots of topics related to how we write at WordPress — here are some that might be of particular interest:

Reviewer 2: I think this guide is more relevant here https://developer.wordpress.org/block-editor/contributors/documentation/copy-guide/

[Style, voice, and tone](https://make.wordpress.org/docs/style-guide/general-guidelines/style-voice-tone/)  
[Language and grammar](https://make.wordpress.org/docs/style-guide/language-grammar/)  
[Accessibility](https://make.wordpress.org/docs/style-guide/general-guidelines/accessibility/)  
[Punctuation](https://make.wordpress.org/docs/style-guide/punctuation/)

> When you have a basic outline of your page and an early version of your content, share a link to your Google Doc in the issue on GitHub.

Reviewer 2: ```suggestion
You can share your progress in an issue as early as a basic outline. 
```
Same theme as earlier, per the [commenting etiquette](https://updateomattic.wordpress.com/2020/07/10/commenting-etiquette-in-the-gutenberg-repository/)
